### PR TITLE
fix(build): add global object "this" to webpack and missing eslint packages

### PR DESCRIPTION
**Pull request by pickymtr**
- Fixed the "Reference error: self is not defined" caused by wepback setting global object to "self" instead of "this", which is not working with typescript. This error occurs when you add the package to your repository and try to build in it. 
- Also adding this missing eslint packages needed to commit on this repository.

### CODE REVIEW

**Review by PierrePerrin — APPROVED** [review]

**Review by adascal — APPROVED** [review]

**Review by stupside — APPROVED** [review]

**Comment by adascal** [comment]
@asgvard any ETA on merging this PR? Really like to see it asap

**Comment by asgvard** [comment]
Hi @adascal , can you elaborate more on these issues please? I don't fully understand why do we need these eslint plugins, I can see they are installed already as part of the airbnb eslint preset. They are present in the `package-lock.json`.

Regarding the Webpack issue, I haven't seen this error neither when developing this library, nor in an external project we use it in. Can you provide more details of the scenario when this happens?

File: package.json [hunk 1]
```diff
@@ -50,6 +50,9 @@
     "eslint-config-airbnb-typescript": "^16.1.0",
     "eslint-config-prettier": "^8.4.0",
     "eslint-plugin-import": "^2.25.4",
+    "eslint-plugin-jsx-a11y": "^6.5.1",
+    "eslint-plugin-react": "^7.29.4",
+    "eslint-plugin-react-hooks": "^4.4.0",
```
[the remark this answers is not in the document]
Comment (pickymtr) [reply package.json:55]: This packages were automatically added by my yarn command because they were in the lock file so it's fine.

**Comment by adascal** [comment]
Regarding eslint packages, airbnb recommend install them as dev dependencies anyway, but I think it's a separate issue, more here:
https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb#eslint-config-airbnb-1

**Comment by asgvard** [comment]
Thanks! Aha, I think it's because of the node environment, which is described here: https://webpack.js.org/configuration/output/#outputglobalobject

Regarding eslint plugins, they are all listed in "peerDependencies" of Airbnb eslint preset, however they do recommend to install them explicitly: https://www.npmjs.com/package/eslint-config-airbnb

I'll merge it tomorrow.

**Comment by adascal** [comment]
Awesome! Thanks

**Review by PierrePerrin — APPROVED** [review]

**Review by asgvard — APPROVED** [review]

**Pull request merged**